torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -271,20 +272,25 @@ def aot_dispatch_autograd(
                     fw_metadata, inner_meta
                 )
             )
+            num_tokens = len(fw_metadata.tokens)
             num_mutated_inp_runtime_indices = len(mutated_inp_runtime_indices)
             num_inner_fwd_outputs = (
                 num_mutated_inp_runtime_indices
                 + inner_meta.num_outputs
                 + inner_meta.num_intermediate_bases
                 + inner_meta.num_outputs_rng_offset
-                + len(
-                    fw_metadata.tokens
-                )  # See Note [Side-Effectful Tokens in AOTAutograd]
+                + num_tokens  # See Note [Side-Effectful Tokens in AOTAutograd]
             )
             fw_module, bw_module = aot_config.partition_fn(
                 fx_g, joint_inputs, num_fwd_outputs=num_inner_fwd_outputs
             )
 
+            # See Note [Side-Effectful Tokens in AOTAutograd]
+            if num_tokens != 0 and aot_config.backend_name == "inductor":
+                unlift_tokens(fw_module, fw_metadata)
```

Comment:
Why do we call `unlift_tokens` multiple times? I see one in dispatch_and_compile_graph too

Reply:
yup!